Substake lock up duration impact on supply, rewrite avoiding inflation rate concept (#3)
nucypher/mining-paper: 2_phase_minting_schedule/main.tex

\author{arjunhassard}
\date{March 2020}
\begin{document}
\maketitle

\section{Summary}

We propose a two-phase subsidy model to bootstrap the NuCypher network, the first phase comprising a \textit{stable issuance}, in which approximately 366 million new tokens are minted and distributed to active (staking) node operators every year in the first phase, set to 5 years. This is equivalent to expanding the supply by 1,002,509 tokens in each of the first 1825 periods (days). In the second phase, commencing on the first period of the fifth year, fewer tokens will be minted in each successive period, such that the daily minting figure decays exponentially with a half-life of 2 years. Over the course of the network's lifetime – as $t$ approaches $\infty$ – the circulating supply will approach a maximum of roughly 3.885 billion tokens. 

\begin{figure}[h!]
    \includegraphics[width=\textwidth]{Two_phase_model.png}
    \caption{Illustration comparing a two-phase model with a common subsidy model, over the first 6 years of the network's lifetime. Note that the y-axis refers to an absolute figure, not a rate.}
    \label{fig:tp}
\end{figure}

> Monetary policy designed for decentralized networks often make reference to a so-called `inflation rate'. In this model, phase one sees an \textbf{identical sum of tokens} minted in each period, so the effective `rate' at which the circulating supply grows decreases period-to-period (and year-to-year). In phase two, the number of tokens minted per day also decreases, which accelerates the decline in the supply growth rate. To avoid misleading terminology, the number of total tokens minted per day shall be hereafter referred to as the \textit{issuance}. Note that the effective growth rate of any given stake also decreases over time, the extent to which is subject to discretionary factors such as the percentage of tokens re-staked by the node operator. 

Reviewer: ```suggestion
Monetary policy designed for decentralized networks often make reference to a so-called `inflation rate'. In this model, phase one sees an \textbf{constant amount of tokens} minted in each period, so the effective `rate' at which the circulating supply grows decreases period-to-period (and year-to-year). In phase two, the number of tokens minted per day also decreases, which accelerates the decline in the supply growth rate. To avoid misleading terminology, the number of total tokens minted per day shall be hereafter referred to as the \textit{issuance}. Note that the effective growth rate of any given stake also decreases over time, the extent to which is subject to discretionary factors such as the percentage of tokens re-staked by the node operator. 
```

\\\\
In the NuCypher protocol, daily issuance is modified by the average stake lock-up duration, but for the sake of simplicity this paper initially discusses a scenario in which all node operators choose a lock-up of 12 months or longer. In reality, the average stake lock-up duration is likely to be lower, which would imply fewer tokens minted per day. This has an impact on the global economy and supply, but regardless, all node operators choose discretionary lock-up durations and receive commensurate subsidies.
\\\\
The second section introduces reasoning for the two-phase model pertaining to network demand, the protocol and node operation, the third introduces an empirical analysis of existing networks, and the final section derives parameters for the new minting schedule.

\section{Motivation \& Rationale}

\subsection{Demand uncertainty}

These demand-related premises form the basis for the two-phase model: 
\begin{enumerate}

> \item There is little certainty regarding the length of time that will elapse between the network's launch and the first trickle of demand from customers (developers, end-users \& others).

Reviewer: ```suggestion
\item There is little certainty regarding the length of time that will elapse between the network's launch and the development of meaningful demand from customers (developers, end-users \& others).
```

\item This early-stage demand is particularly fragile, because new customers are naturally more fickle. Specifically, neither developers nor their end-users have compelling reasons to tolerate an inadequate service (e.g. too few stakers available to service a sharing policy), such as the sunk costs of integration, reliance on the service, or other forms of path dependency, all of which are more likely to play a role later in the network's lifetime. New customers also take a greater risk than later adopters, as they do not have the experience of existing customers as evidence of a reliable or valuable service.
\item A popular subsidy model involves daily token issuance dropping exponentially, right from launch. If the fee market does not sufficiently mature in time to make up for the sharp reduction in earnings the early years of the network's existence, some node operations may be rendered entirely unsustainable. 

> \item Most importantly, some of NuCypher's early customers will (inflexibly) require long-lasting sharing policies. This means that the protocol must motivate the successive re-commitment to long token lock durations (i.e. staking for 12+ months) by a sufficient population of node operators. A diminished subsidy issuance, in lieu of a mature fee market, may not be enough to incentivize this repeated and essential re-commitment. 

Reviewer: ```suggestion
\item Most importantly, some of NuCypher's early customers may (inflexibly) require long-lasting sharing policies. This means that the protocol must motivate the successive re-commitment to long token lock durations (i.e. staking for 12+ months) by a sufficient population of node operators. A diminished subsidy issuance, in lieu of a mature fee market, may not be enough to incentivize this repeated and essential re-commitment. 
```

Reviewer 2: ^ Ignore above, it was an old change, before you made updates based on maclane's suggestions.

A stable issuance in the first phase brings other benefits, including the partial mitigation of a centralizing dynamic suffered by many staking networks, in which larger node operators utilize subsidies to increase/consolidate control of the token supply, thanks to the allocation of tokens in proportion to stake size. Moreover, larger operators can re-stake a greater percentage of subsidies (not needing to convert as much to fiat to cover operational costs), precipitating a `rich-get-richer' dynamic. If token issuance is highest in the earliest days of the network, larger operators are granted an even greater head-start over new node operators who start staking in a later era, \textbf{the presence of whom provides the network with an important, long-term counter-balance against centralization}. Relatedly, an important rationale for the \textit{work token} model is the supposition that new node operators purchase the network's native token in order to stake and earn revenue. With a token issuance that decays from genesis, new operators are doubly disadvantaged – they must pay the market price to acquire tokens, plus they receive a diminished subsidy. 
\\\\

> Hence, a two-phase minting schedule assists all would-be node operators who don't happen to stake right at network launch, but decide to do so at a later date. By addressing the temporal inequity of the previous draft, we impede problematic centralization trends.

Reviewer: ```suggestion
Hence, a two-phase minting schedule assists all would-be node operators who don't happen to stake right at network launch, but decide to do so at a later date. By addressing the temporal inequity of immediately decreasing issuance, we impede problematic centralization trends.
```

\subsection{Risks to operational feasibility, incentives and pricing}

> As mentioned, the two-phase model implies, relative to other subsidy models, a lower number of tokens minted on day one. This in itself confers further advantages to network health:  

Reviewer: This is not really clear since even with two-phase inflation, the amount of tokens minted on day one is the highest it will ever be.

Author: ```suggestion
As Figure 1 shows, a `generic' subsidy model involves high issuance at genesis. In the two phase model, the number of tokens minted per day isn't only stable in phase one, it also commences and remains a lower figure for the first few years. This difference confers further benefits to network health:  
```
Agree, the use of 'day one' is unnecessary.

> As mentioned, the two-phase model implies, relative to other subsidy models, a lower number of tokens minted on day one. This in itself confers further advantages to network health:  

Author: ```suggestion
As Figure 1 shows, a `generic' subsidy model involves high issuance at genesis. In the two phase model, the number of tokens minted per day isn't only stable in phase one, it also commences and remains a lower figure for the first few years. This difference confers further benefits to network health:  
```
Agreed, the use of 'day one' is unnecessary. @mswilkison

Author: I think what I was originally trying to convey here wasn't v clear
```suggestion
As illustrated in Figure 1, `generic' subsidy models tend to involve a high issuance at genesis and in the first few years of the network's existence. Setting the peak issuance (the maximum number of tokens minted per day) at a comparatively lower figure addresses the following potential problems: 
```

Reviewer 2: 👍

\begin{itemize}
\item The greater the real-world value of the subsidy, the further from a fee-driven `reality' a node's set-up may be in economic terms (for example, in terms of operational efficiency). When subsidies eventually run out, node operators who are unable to adapt may cease operations, threatening general service availability or even reneging on existing commitments to users. Under a model with higher token issuance early on, these inefficient operators may control a substantial proportion of the circulating supply by this point. 

> \item If subsidies are worth significantly more than the typical sum of transaction fees earned in the same period, then the incentive to earn those fees may be blunted. This is particularly problematic for the protocol if the node operator duty required to collect subsidies does not align well with the duty required to earn fees, and the reliability of the latter impacts overall service quality. 

Author: Duty does sound a little pompous! Work is fine, it just doesn't strongly emphasise the distinction between various staker responsibilities, which is the key point here

    3.885 = 1 + 5 \cdot  I_{stable} + I_{stable} \cdot \left[\frac{-2^{(1-\frac{t}{2})}}{\ln{2}}\right]_0^\infty
\end{equation}

We are now able to find an expression for $I_{stable}$. Once again note that this variable is expressed as a percentage of the genesis supply, $S_0$.

\begin{equation}
    2.885 = I_{stable} \cdot (5 + \frac{2}{\ln{2}})
\end{equation}

\begin{equation}
   I_{stable} = \frac{2.885}{7.885} = 36.6\%
\end{equation}

We use $I_{stable}$ to find the number of tokens minted per year for the first 5 years.

\begin{equation}
   I_{stable} \cdot S_0 = 365,915,960\:\:new \:tokens \:per \:year
\end{equation}

This is equivalent to 1,002,509 newly minted tokens released into the circulating supply, every period (day) for the first 1825 periods. After this, the issuance decays on a period-to-period basis. For example, 1,001,558 tokens are minted on day 1827, which is 951 fewer than the previous day.
\\\\
In this token minting schedule, we can work out that approximately 1.83 billion tokens will be minted in the first phase, followed by a further 1.056 billion between the commencement of phase two and t = $\infty$.

\subsection{Variable stake lock-up duration}

> In the NuCypher protocol, the number of tokens distributed to node operators in a given period depends on the stage in the minting schedule we are at and the size of their stake, but is also a function of the remaining duration their stake is locked up for (in fact, the duration of each of the `sub-stakes' they control). Any sub-stake locked up for a short duration receives less than the maximum subsidy, thereby slowing the expansion of the entire circulating supply – tokens are only minted if they are to be distributed to node operators. The number of tokens minted daily for a given sub-stake depends on the coefficient $\kappa$:

Reviewer 2: "depends on the [current] stage in the minting schedule and..."
"also a function of the remaining [locked duration of their stake] (in fact..."
"Any sub-stake [locked] for a ..."

```suggestion
In the NuCypher protocol, the number of tokens distributed to node operators in a given period depends on the current stage in the minting schedule and the size of their stake, but is also a function of the remaining locked duration of their stake (in fact, the duration of each of the `sub-stakes' they control). Any sub-stake locked for a short duration receives less than the maximum subsidy, thereby slowing the expansion of the entire circulating supply – tokens are only minted if they are to be distributed to node operators. The number of tokens minted daily for a given sub-stake depends on the coefficient $\kappa$:
```

Where $S_{P2}$ is the maximum number of tokens to be minted in the second phase (approximately 1.056 billion tokens). 
\\\\
The circulating supply at a given period is the sum of the tokens received thus far by all sub-stakes: 
\begin{equation}
    dS_p= \sum_i ds_{i,p},
\end{equation}

The implication is that it will likely take longer than 5 years to mint all the tokens reserved for phase one. The year in which the protocol switches from the first to second phase, $T_{1\rightarrow2}$, is therefore : 

\begin{equation}
    T_{1\rightarrow2}= \frac{S_{P1}}{S_0 \cdot \kappa^* \cdot I_{stable}}
\end{equation}

Where $\kappa^*$ is the average lock duration of all sub-stakes, weighted by relative size, across all periods in question. For example, if $\kappa^*$ is 73 days, then the minting schedule will switch phases about 8 1/3 years after network launch. If $\kappa^*$ is 11 months, $T_{1\rightarrow2}$ equals 5 years \& 79 days. 
\\\\
For reference, the supply in the second phase can be calculated with this equation: 

\begin{equation}
    S(t) = S_0 \left[1 + \frac{I_{stable} \cdot T_{1/2}}{\ln{2}}\left(1 - 2^{-\frac{t \cdot \kappa^*}{T_{1/2}}} \right) \right].
\end{equation}

> Which implies that the rate of issuance decay is inversely proportional to $\kappa^*$. In a similar fashion to phase one, shorter average lock-ups slow down the minting of tokens. 

Reviewer 2: Is the inverse proportional relationship of "rate" & "kappa" correct here? We should be careful using the word "rate" here.
If kappa is low, i.e. short durations, then the decay rate slows i.e. lowers (and vice versa)

Do you instead mean that the **half-life** (i.e. the amount of time for the quantity to decrease by half) is inversely proportional to kappa?

Author: Well-spotted! It's proportional, and I agree about 'rate' (the term that shall not be named). 

The reason I didnt say that kappa* is inversely proportional to the 'half-life' is that we previously say the half-life _is T_1/2 _. Kappa doesn't affect T_1/2  – as you can see from eq. 13, it modifies the expression for decay over time as a whole: -2^(-t * k / T_1/2) 

```suggestion
Which implies that issuance decay is proportional to $\kappa^*$. In a similar fashion to phase one, shorter average lock-ups slow down the minting of tokens. 
```

Reviewer 2: 👍 - Yes decay and not T_1/2 - I misread "8 1/3 years" and "5 years \& 79 days".
